dist/cms.js

```diff
@@ -151,12 +151,57 @@ function getWordPressContentType() {
   return content;
 }
 
+/**
+ * Obtains data about Interactivity API usage on the page.
+ *
+ * @returns {object} Object with fields `total_regions` and `total_regions_by_namespace`.
+ */
+function getInteractivityAPIUsage() {
+  // Look for data-wp-interactive regions.
+  const interactiveRegions = document.querySelectorAll('[data-wp-interactive]');
+
+  // Count the regions by namespace.
+  const regionNamespaceCounts = [];
+  interactiveRegions.forEach( region => {
+    // Extract the namespace from JSON or as a string
+    let namespace = '';
+    const regionAttribute = region.getAttribute('data-wp-interactive');
+    try {
+      const regionData = JSON.parse( regionAttribute );
+      if ( regionData.namespace && 'string' === typeof regionData.namespace ) {
+        namespace = regionData.namespace;
+      }
+    } catch ( e ) {
+      namespace = regionAttribute;
+    }
+    if ( '' !== namespace ) {
+      regionNamespaceCounts[ namespace ] = ( regionNamespaceCounts[ namespace ] || 0 ) + 1;
+    }
+  })
+
+  return {
+    total_regions: interactiveRegions.length,
+    total_regions_by_namespace: regionNamespaceCounts,
+  };
```

Comment:
```suggestion
  return JSON.stringify({
    total_regions: interactiveRegions.length,
    total_regions_by_namespace: regionNamespaceCounts,
  });
```

Reply:
@tunetheweb Why is this necessary? I personally would find it odd to find stringified JSON within data that's already JSON. That's always been a bit mysterious to me with HTTP Archive data (also in `payload` column), why do we have to JSON-decode twice?

Reply:
Same question. The github-action did return the expected data:

![image](https://github.com/HTTPArchive/custom-metrics/assets/2676022/3a1c5db6-ad8b-447c-9a0d-55492adc2e70)

Reply:
+1 to what @adamsilverstein shared. There isn't actually an array in the data, it's only objects.

But even generally, I'm not sure what's the limitation of using `JSON.stringify()` on an array? 🤔

Reply:
>Why is this necessary? I personally would find it odd to find stringified JSON within data that's already JSON. 

Yes as mentioned below you only need to stringify the outermost return. Apologies for the confusion.

> That's always been a bit mysterious to me with HTTP Archive data (also in payload column), why do we have to JSON-decode twice?

Ah that's legacy thing. Some of the older custom metrics are double escaped. Yeah I don't like that either. We shouldn't do that anymore for newer custom metrics. Dunno why we did to be honest - before my time.

> +1 to what @adamsilverstein shared. There isn't actually an array in the data, it's only objects.

It was (incorrectly) originally set to be an Array as mentioned in my comment above.

> But even generally, I'm not sure what's the limitation of using JSON.stringify() on an array? 🤔

As described below the "limitation" is when you set up an array, then using "named" array entries, which aren't actually array entries at all:

```js
const myArray = []
myArray[1] = 'value'; // this is fine
myArray['key'] = 'value'
```

That last line is incorrect and actually is shorthand for `myArray.key = 'value';'
You've basically set an attribute on the array and not added a value to the array.
When you convert the array to a string, it doesn't include attributes (it would if it was an object, but it's not, it's an array). It only exports entries to the string and in this example `myArray` only has one entry not two (as can be seen with `myArray.length`).

Why does JavaScript allow you to set attributes on an Array when it's almost certainly done in error? 🤷‍♂️ Ultimately everything in JavaScript is an object so you can always set attributes.

Why does JavaScript have a similar syntax for setting array values and array attributes? 🤷‍♂️🤷‍♂️ Just one of the many fun things with JavaScript!